### PR TITLE
add Gnome

### PR DESCRIPTION
**Pull request by ThyWoof**

### CODE REVIEW

File: SolastaCommunityExpansion/Models/RacesContext.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+using System.Reflection;
+using static SolastaModApi.DatabaseHelper;
+
+namespace SolastaCommunityExpansion.Models
+{
+    public class RacesContext
+    {
+        internal static CharacterRaceDefinition GnomeRace { get; private set; } = BuildGnome();
+
+        private static T MemberwiseClone<T>(T base_object)
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Models/RacesContext.cs:13]: I don't think this is doing what you expect.  It's a shallow copy so you may find modifying collections in the copy modify the original.

Try this in LinqPad

```
void Main()
{
	var t = new Test { Number = 1, Items = new List<string> { "one", "two" }};

	t.Number.Dump("orig");
	string.Join(",", t.Items).Dump("orig");

// take copy
	var t2 = MemberwiseClone(t);

// modify original
	t.Number = 5;
	t.Items.Add("three");

// we see change to Items
	t2.Number.Dump("copy");
	string.Join(",", t2.Items).Dump("copy");
}

class Test
{
	public int Number {get;set;}
	public List<string> Items {get;set;}
}
// Define other methods and classes here
private static T MemberwiseClone<T>(T base_object)
{
	MethodInfo memberwiseCloneMethod = typeof(T).GetMethod("MemberwiseClone", BindingFlags.Instance | BindingFlags.NonPublic);
	return (T)memberwiseCloneMethod.Invoke(base_object, null);
}
```

File: SolastaCommunityExpansion/Models/RacesContext.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
+﻿using SolastaCommunityExpansion.Builders;
+using SolastaCommunityExpansion.Builders.Features;
+using SolastaModApi.Extensions;
+using System.Reflection;
+using static SolastaModApi.DatabaseHelper;
+
+namespace SolastaCommunityExpansion.Models
+{
+    public class RacesContext
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Models/RacesContext.cs:9]: public static

File: SolastaCommunityExpansion/Models/RacesContext.cs (hunk 1 above)
Comment (PhilPJL) [reply SolastaCommunityExpansion/Models/RacesContext.cs:13]: See https://docs.microsoft.com/en-us/dotnet/api/system.object.memberwiseclone?view=net-6.0

Comment (PhilPJL) [reply SolastaCommunityExpansion/Models/RacesContext.cs:13]: Maybe Object.Instantiate would work as a deep copy.

**Review by PhilPJL — CHANGES_REQUESTED** [review]
Clone method is faulty

File: SolastaCommunityExpansion/Models/RacesContext.cs (hunk 1 above)
Comment (ThyWoof) [reply SolastaCommunityExpansion/Models/RacesContext.cs:13]: this came from migration. Object.Instantiate won't work on this class. Any suggestions?

Comment (PhilPJL) [reply SolastaCommunityExpansion/Models/RacesContext.cs:13]: Manually create a copy method.

var orig = CharacterRaceDefinitions.Halfling.RacePresentation;
var rp = new RacePresentation();

rp.SetMaleBeardShapeOptions(orig.MaleBeardShapeOptions);
... etc

Comment (ThyWoof) [reply SolastaCommunityExpansion/Models/RacesContext.cs]: implemented https://github.com/Burtsev-Alexey/net-object-deep-copy

**Review by PhilPJL — APPROVED** [review]
Only issues I can think of are
1) the gnome and warlock images are high res and take up about 2Mb. The final .dll is <5Mb  total including the images.  Maybe they could be shrunk down.
2) was concerned the new .Copy could conflict with other .Copy methods but it doesn't seem to.  Maybe rename it DeepCopy?

**Review by PhilPJL — CHANGES_REQUESTED** [review]
I'm thinking Copy methods should be called DeepCopy to avoid accidentally using this Copy when there's a Copy extension method on a blueprint.

Also DeepCopy should throw if attempting to copy a UnityEngine.Object.  A DeepCopy of a blueprint/Unity object may cause problems and we should use Object.Instantiate etc in those cases.

**Comment by ThyWoof** [comment]
renamed to DeepCopy and added check on unity objects

**Review by PhilPJL — APPROVED** [review]

**Pull request merged**